taf/updater/updater_pipeline.py

```diff
@@ -402,86 +402,101 @@ def set_last_validated_commit(self, validation_repo):
 
     def check_if_local_repositories_clean(self):
         try:
-            # check if the auth repo is clean first
-            if self.state.existing_repo:
-                auth_repo = AuthenticationRepository(
-                    path=self.auth_path, urls=[self.url]
+            # Early exit if the repository does not exist
+            if not self.state.existing_repo:
+                return UpdateStatus.SUCCESS
+
+            # Check the main authentication repository
```

Comment:
Very minor, but with nothing. A pipeline was designed to update one auth repo and a list of its targets (data) repos. So main auth repo could be confusing here as there is only one auth repo. Update of dependencies is handled in updater.py